notes.go

```diff
@@ -136,7 +136,33 @@ func (srv *notesAPI) DeleteNote(ctx context.Context, req *notesv1.DeleteNoteRequ
 }
 
 func (srv *notesAPI) ListNotes(ctx context.Context, req *notesv1.ListNotesRequest) (*notesv1.ListNotesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "unimplemented")
+	token, err := srv.authenticate(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validators.ValidateListNoteRequest(req)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Check user is part of the group.
+	_, err = srv.groups.GetGroup(ctx, &models.OneGroupFilter{GroupID: req.GroupId}, token.AccountID)
+	if err != nil {
+		return nil, statusFromModelError(err)
+	}
+
+	notes, err := srv.notes.ListNotesInternal(ctx, &models.ManyNotesFilter{AuthorAccountID: req.AuthorAccountId}, &models.ListOptions{Limit: 20, Offset: 0})
```

Comment:
Hardcoding `&models.ListOptions{Limit: 20, Offset: 0}` makes it that any query that matches >20 notes will only return the first 20 notes. For example, a group with 100 notes will appear as though it only has 20.

Also, you should pass the `GroupId` to the `ListNotesInternal` method. Right now you're only `AuthorAccountId` as parameter.

We should add a `limit` and `offset` to the `ListNotesRequest` message in the `protorepo` and pass it as parameter here.

Reply:
Also when making the call to the database (through `ListNotesInternal`) we should make sure to **project** (MongoBD projection option) the results so that only the "metadata" of the note is returned.

Basically, if I list a hundred notes, I'm usually only interested in the name, title and author of those notes, not the list of blocks of the 100 notes because, theoretically, that could be a really big response object. If each note has 100 blocks, a single call to `ListNotes` could fetch up to 10,000 blocks, which is too much for a single JSON response.

Reply:
Ask @edouard-sn because he's way better than me at projections. Tell him to not use aggregation pipelines though but instead use the `options.SetProjection()` option of the MongoDB driver for calls to `coll.Find`.

Reply:
Also could you do the same for groups? The `ListGroups` endpoint should not return the group's members, conversations, invites and inviteLinks arrays for the reasons stated above.

Reply:
i'm not that good at it lol but here's what i found if that can make things faster for you @gabriel654165 
```go
// 1 To include, 0 to exclude, if you only use 1s you don't have to specify the ones you exclude (0s) 
// and same if you only use 0s
opts := options.Find().SetProjection(bson.D{{"name", 1}, {"description", 1}})  // Will only select name and description field
//...
cursor, err := coll.Find(context.TODO(), filter, opts)
```

https://www.mongodb.com/docs/drivers/go/current/fundamentals/crud/read-operations/project/

Reply:
Amazing! We're looking to keep all the fields except the array fields (so `_id`, `name`, `description`, `workspaceAccountId`, etc).

Reply:
Also completely unrelated @gabriel654165 , but could you look into why the CI is failing? It seems maybe we're missing a `go mod tidy` or some dependencies.